packages/@ember/-internals/glimmer/lib/helpers/array.ts

```diff
@@ -42,6 +42,8 @@ import { ReifyPositionalReference } from '@glimmer/runtime';
    @public
  */
 
-export default function(args: VMArguments): PathReference {
-  return new ReifyPositionalReference(args.positional.capture());
+export default function(args: VMArguments): Reference<unknown[]> {
+  let captured = args.positional.capture();
+
+  return createComputeRef(() => reifyPositional(captured), null, 'array');
```

Comment:
This is because `action` returns an UnboundReference, which is unusual. Most helpers return a `ComputeRef`, which adds the `result of a...` portion in Glimmer VM. I did it this way so that we wouldn't be repeating `result of a...` for every single helper, which would add byte size and maintenance cost.

Agreed that we should come up with a better system in the future though, I do think the way that helpers are defined should be more consistent.